src/main/scala/intellij/haskell/external/repl/ProjectStackReplProcess.scala

```diff
@@ -47,20 +48,24 @@ private[repl] class ProjectStackReplProcess(project: Project) extends StackReplP
   }
 
   private def load(psiFile: PsiFile, moduleName: Option[String]): Option[(StackReplOutput, Boolean)] = {
-    val filePath = getFilePath(psiFile)
-    val moduleNameAction = ApplicationManager.getApplication.runReadAction(new Computable[Option[String]] {
-      override def compute(): Option[String] = {
-        moduleName.orElse(HaskellPsiUtil.findModuleName(psiFile))
+    if (!HaskellConsoleView.isConsole(psiFile)) {
```

Comment:
Your initial original solution does not handle the cause, it's a work around to prevent commands are executed on the project repl. We have to handle it earlier in the program flow.